test/unit/ajax.js

```diff
@@ -1523,6 +1523,93 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - responseURL", 13, function( assert ) {
+
+		// This function generates complete
+		// absolute URL without any relative path components
+		// Example:
+		// Input: http://localhost:9876/base/test/data/testinit.js/../../../test/data/mock.php?action=responseURL&155821644568982930
+		// Output: http://localhost:9876/base/test/data/mock.php?action=responseURL&155821644568982930
+		function normalizeUrl( url ) {
+			var finalUrl = url;
+			while ( finalUrl.indexOf( "/../" ) > -1 ) {
+				finalUrl = finalUrl.replace( /\/[^/]+\/\.\.\//g, "/" );
+			}
+			return finalUrl;
+		}
+
+		var successURL = url( "mock.php?action=responseURL" ),
+			errorURL = "http://example.invalid",
+			redirectAndSuccessURL = url( "mock.php?action=responseURL&url=" + encodeURIComponent( successURL ) ),
+			redirectAndErrorURL = url( "mock.php?action=responseURL&url=" + encodeURIComponent( errorURL ) ),
+			jsonpURL = baseURL + "mock.php?action=jsonp&callback=?",
```

Comment:
Is there any reason why this uses `baseURL` directly instead of the `url` function?